src/main/scala/io/delta/DeltaTable.scala

```diff
@@ -97,15 +99,248 @@ class DeltaTable (df: Dataset[Row]) extends DeltaTableOperations {
     executeDelete(None)
   }
 
+
   /**
    * :: Evolving ::
    *
-   * Merge data from the `source` table that match the given `condition`
+   * Update data from the table based on the rules defined by `set`.
    *
-   * @param source source Dataframe to be merged.
-   * @param condition Boolean SQL expression
-   * @return
+   * Scala example to increment the column `data`.
+   * {{{
+   *    import org.apache.spark.sql.functions._
+   *
+   *    deltaTable.update(Map("data" -> col("data") + 1))
+   * }}}
+   *
+   * @param set rules to update a row as a Scala map between target column names and
+   *            corresponding expressions as Column objects.
+   * @since 0.3.0
+   */
+  @Evolving
+  def update(set: Map[String, Column]): Unit = {
+    executeUpdate(set, None)
+  }
+
+  /**
+   * :: Evolving ::
+   *
+   * Update data from the table based on the rules defined by `set`.
+   *
+   * Java example to increment the column `data`.
+   * {{{
+   *    import org.apache.spark.sql.Column;
+   *    import org.apache.spark.sql.functions;
+   *
+   *    deltaTable.update(
+   *      new HashMap<String, Column>() {{
+   *        put("data", functions.col("data").plus(1));
+   *      }}
+   *    );
+   * }}}
+   *
+   * @param set rules to update a row as a Java map between target column names and
+   *            corresponding expressions as Column objects.
```

Comment:
between target column names and their corresponding update expressions as Column objects